Src/IronPython/Runtime/Descriptors.cs

```diff
@@ -30,10 +30,17 @@ namespace IronPython.Runtime {
     public class staticmethod : PythonTypeSlot {
         internal object _func;
 
-        public staticmethod(object func) {
-            this._func = func;
+        public staticmethod(CodeContext/*!*/ context, object func) {
+            __init__(context, func);
         }
 
+        public void __init__(CodeContext/*!*/ context, object func) {
+            if (!PythonOps.IsCallable(context, func))
```

Comment:
It looks like CPython doesn't have this restriction on initialization (similarly for classmethod)